### ISSUE TITLE
`save` (without -r) sub-superdataset takes too long - checks its subdatasets

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#4523]
Sorry if I have failed to find a previous issue - checked opened and some closed with `performance` but have not spotted one.

Use case: `openneuro` subdataset which contains a good (>300) number of subdatasets but otherwise not burdened with too many files (about 900).  Just to save an updated state of that dataset without doing recursive save, takes over a minute.  Well -- in this case there even no updated state. I am explicitly pointing to current superdataset (`-d .`) and asking just to save possibly updated state of openneuro.  There is no `-r` : 
```shell
$> datalad --version; time datalad save -d . -m 'there is no changes' openneuro
datalad 0.12.6
datalad save -d . -m 'there is no changes' openneuro  42.56s user 42.15s system 102% cpu 1:22.57 total

$> time git -C openneuro status
On branch master
nothing to commit, working tree clean
git -C openneuro status  1.71s user 3.18s system 134% cpu 3.640 total

$> source ~datalad/datalad-master/venvs/dev3/bin/activate                                                                                    

$> datalad --version; time datalad save -d . -m 'there is no changes' openneuro
datalad 0.13.0rc1.dev19
datalad save -d . -m 'there is no changes' openneuro  42.79s user 40.78s system 102% cpu 1:21.86 total
```

So why does it need to then inspect every subdataset of the openneuro?  Here is what I see  with `-l DEBUG`:
```
[DEBUG  ] Done query repo: ['git', 'ls-tree', 'HEAD', '-z', '-r', '--full-tree', '-l'] 
[DEBUG  ] Done GitRepo(/mnt/datasets/datalad/crawl/openneuro/ds000105).get_content_info(...) 
[DEBUG  ] GitRepo(/mnt/datasets/datalad/crawl/openneuro/ds000107).get_content_info(...) 
[DEBUG  ] Query repo: ['git', 'ls-files', '--stage', '-z', '-d', '-m', '--exclude-standard', '-o'] 
[DEBUG  ] Done query repo: ['git', 'ls-files', '--stage', '-z', '-d', '-m', '--exclude-standard', '-o'] 
[DEBUG  ] Done GitRepo(/mnt/datasets/datalad/crawl/openneuro/ds000107).get_content_info(...) 
[DEBUG  ] GitRepo(/mnt/datasets/datalad/crawl/openneuro/ds000107).get_content_info(...) 
```
is it something we could fix or it is just my misunderstanding again of normal `save` operation and I am missing a use-case where inspection of subdatasets would be needed for some reason?

**Comment by mih** [comment]
I think it can and should be fixed. I did not investigate in detail, but something along these lines might already be sufficient:

```diff
diff --git a/datalad/core/local/save.py b/datalad/core/local/save.py
index 8ccfcdc10..7e7458b42 100644
--- a/datalad/core/local/save.py
+++ b/datalad/core/local/save.py
@@ -202,6 +202,7 @@ class Save(Interface):
                 dataset=dataset,
                 path=path,
                 untracked=untracked_mode,
+                eval_submodule_state='full' if recursive else 'commit',
                 report_filetype=False,
                 recursive=recursive,
                 recursion_limit=recursion_limit,
```

**Comment by yarikoptic** [comment]
I guess you had missed #4526

**Comment by yarikoptic** [comment]
closed by 4f0876e18dcc0b428844f7278511cf45a3a8428f
